### PR TITLE
Diagnostic Logging for Azure Resources

### PR DESCRIPTION
**Pull request by ronaldheft-usds**
This PR does adds logging via an event hub to pass to Splunk for several Azure resources.

## Changes
- Configures storage account blob access logging for Splunk (#442)
- Configures function app logging for Splunk (#443)
- Configures database logging for Splunk (#444)

With all of the above changes, all active resources using in prod will have logs that can be sent to Splunk.

## Checklist
- [x] Tested locally? Tested against my resource group.
- [ ] Added new dependencies?
- [ ] Updated the docs?
- [ ] Added a test?
- [ ] Did you check for sensitive data, and remove any?
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

## Fixes 

- #issue

## To Be Done

- Securely communicate all of the access keys to Splunk teams once these roll out to test and production.

### CODE REVIEW

**Comment by ronaldheft-usds** [comment]
Updated to fix merge conflict.

File: operations/app/src/modules/storage/main.tf [hunk 1]
```diff
@@ -23,6 +23,120 @@ resource "azurerm_storage_account" "storage_account" {
...
+    enabled = false
+
+    retention_policy {
+      days = 0
+      enabled = false
+    }
+  }
+}
+
+module "storageaccount_access_log_event_hub_log" {
```
Comment (cglodosky) [inline operations/app/src/modules/storage/main.tf:83]: Is this module supposed to be in here twice?  It looks identical to the first one.

File: operations/app/src/modules/storage/main.tf [hunk 2]
```diff
@@ -23,6 +23,120 @@ resource "azurerm_storage_account" "storage_account" {
...
+  source = "../event_hub_log"
+  resource_type = "storage_account"
+  log_type = "access"
+  eventhub_namespace_name = var.eventhub_namespace_name
+  resource_group = var.resource_group
+  resource_prefix = var.resource_prefix
+}
+
+resource "azurerm_monitor_diagnostic_setting" "storageaccount_access_log" {
```
Comment (cglodosky) [inline operations/app/src/modules/storage/main.tf:92]: Is this resource supposed to be in here twice?  It looks identical to the first one.

File: operations/app/src/modules/storage/main.tf (hunk 1 above)
Comment (ronaldheft-usds) [reply operations/app/src/modules/storage/main.tf]: I think the duplicate was introduced in the merge conflicts I fixed this morning. I removed the duplicated lines.

File: operations/app/src/modules/storage/main.tf (hunk 2 above)
Comment (ronaldheft-usds) [reply operations/app/src/modules/storage/main.tf]: I think the duplicate was introduced in the merge conflicts I fixed this morning. I removed the duplicated lines.

**Comment by cglodosky** [comment]
Looks good!  It will be nice to start capturing these logs.

**Review by cglodosky — APPROVED** [review]

**Pull request merged**